client/src/components/mixins/nr-affiliation-mixin.ts

```diff
@@ -0,0 +1,68 @@
+import Vue from 'vue'
+import { Component } from 'vue-property-decorator'
+import AuthServices from '@/services/auth.services'
+import BusinessServices from '@/services/business.services'
+import newReqModule from '@/store/new-request-module'
+import { BusinessRequest, CreateNRAffiliationRequestBody } from '@/models/business'
+import { NameRequestI } from '@/models'
+
+@Component
+export default class NrAffiliationMixin extends Vue {
+  /**
+   * Create an affiliation between current authenticated Account and current Nr
+   * @param nr The Nr to affiliate
+   */
+  async createAffiliation (nr: NameRequestI): Promise<any> {
+    try {
+      // show spinner since the network calls below can take a few seconds
+      this.$root.$emit('showSpinner', true)
+
+      const currentOrganizationId = JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT')).id
+      const requestBody: CreateNRAffiliationRequestBody = {
+        businessIdentifier: nr.nrNum,
+        phone: nr.applicants?.phoneNumber || '',
+        email: nr.applicants?.emailAddress || ''
+      }
+
+      // Request to affiliate NR to current account
+      const nrResponse = await AuthServices.createNRAffiliation(currentOrganizationId, requestBody)
+      if (nrResponse?.status === 201) {
```

Comment:
I'm looking at createBusiness() and I see 3 possible return cases:
1. the POST succeeded and the status is 2xx, so return the response
2. the POST succeeded and the status is not 2xx, so return nothing
3. the POST failed, so return `{ errorMsg }`

Based on this, I agree that it's wise to check here that you have case 1. Two questions:
- must you receive a 201, or is any 2xx valid?
- what are you doing to catch errors, since createBusiness() only returns an exception if the second call (removeAffiliation) fails?

Reply:
(Sorry, move this comment to line 48.)

Reply:
I definitely agree, i think i need to expand this condition to include a 200 (in case) as well as throw an error to the catch

Reply:
I see that createNRAffiliation() will either return a resolved or rejected promise (ie, then and catch) (ie, success or exception).

If createNRAffiliation() fails then the catch for the block here will be executed, which is fine.

If createNRAffiliation() succeeds then you check for a 201, which is OK, but what if you receive another "succeess" status? Is that possible?

Reply:
K, pushed a change. 

That being said, we will still need to do work on handling the errors more explicitly, but that will be covered when designs are updated for those dialogs etc.